pymc/sampling/forward.py

```diff
@@ -1083,3 +1090,122 @@ def vectorize_over_posterior(
                 f"The following random variables found in the extracted graph: {remaining_rvs}"
             )
     return vectorized_outputs
+
+
+def loop_over_posterior(
```

Comment:
instead of a separate function have an argument: `use_scan: bool = False` in the previous function? Most logic should be the same. 

This flatten batch -> scan -> reshape logic should go in PyTensor, it's needed in general